crates/ruma-events/src/media_preview_config.rs

```diff
@@ -0,0 +1,68 @@
+//! Types for the [`m.media_preview_config`] event.
+//!
+//! [`m.media_preview_config`]: https://github.com/matrix-org/matrix-spec-proposals/pull/4278
+
+use ruma_macros::StringEnum;
+use serde::{Deserialize, Serialize};
+
+use crate::{macros::EventContent, PrivOwnedStr};
+
+/// The payload for `MediaPreviewConfigEventContent`.
+#[cfg(feature = "unstable-msc4278")]
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "m.media_preview_config", kind = GlobalAccountData)]
+pub struct MediaPreviewConfigEventContent {
+    /// The media previews configuration.
+    #[serde(rename = "media_previews")]
+    pub media_previews: MediaPreviews,
+    #[serde(rename = "invite_avatars")]
+    /// The invite avatars configuration.
+    pub invite_avatars: InviteAvatars,
+}
+
+/// The configuration that handles if media previews should be shown in the timeline.
+#[cfg(feature = "unstable-msc4278")]
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, StringEnum)]
+#[non_exhaustive]
+pub enum MediaPreviews {
+    /// Off - Media previews should be hidden.
```

Comment:
I don't think that it's necessary to repeat the name of the variant in the docs. If you want to however you could put the actual string that will appear in the account data, i.e. `off` (using inline code).

Same for the other variants and the other enum below.